packages/vite/src/node/utils.ts

```diff
@@ -421,12 +421,15 @@ export function writeFile(
  * unnecessary in the first place)
  */
 export function isFileReadable(filename: string): boolean {
+  let stat = null
   try {
-    fs.accessSync(filename, fs.constants.R_OK)
-    return true
-  } catch {
-    return false
+    stat = fs.statSync(filename, { throwIfNoEntry: false })
+  } catch (e) {
+    if (e && (e.code === 'ENOENT' || e.code === 'ENOTDIR')) return false
+
+    throw e
   }
+  return !!stat
```

Comment:
I wonder if we can go even further to remove the try..catch altogether. Since `throwIfNoEntry: false`, `ENOENT` would never happen. Not sure `ENOTDIR` would happen too. The `fs.statSync` function seems unlikely to throw errors from my testing.

Reply:
`throwIfNoEntry` option was added in Node 15.3, before that it is a no-op. So `catch` block is needed for older versions of Node. It can probably be simplified to not check for error code, as it seems that Vite did not care about actual file access errors before.

Reply:
Ah thanks for the explanation! The new changes looks good to me.